src/cryptojwt/key_jar.py

```diff
@@ -325,119 +306,136 @@ def match_owner(self, url):
+        if _iss:
+            return _iss[0]
 
         raise KeyError("No keys for '{}' in this keyjar".format(url))
 
     def __str__(self):
         _res = {}
-        for _id, kbs in self.issuer_keys.items():
-            _l = []
-            for kb in kbs:
-                _l.extend(json.loads(kb.jwks())["keys"])
-            _res[_id] = {"keys": _l}
+        for _id, _issuer in self._issuers.items():
+            _res[_id] = _issuer.key_summary()
         return json.dumps(_res)
 
-    def load_keys(self, issuer, jwks_uri='', jwks=None, replace=False):
+    @deprecated_alias(issuer='issuer_id', owner='issuer_id')
+    def load_keys(self, issuer_id, jwks_uri='', jwks=None, replace=False):
         """
         Fetch keys from another server
 
         :param jwks_uri: A URL pointing to a site that will return a JWKS
         :param jwks: A dictionary representation of a JWKS
-        :param issuer: The provider URL
+        :param issuer_id: The provider URL
         :param replace: If all previously gathered keys from this provider
             should be replace.
         :return: Dictionary with usage as key and keys as values
         """
 
-        logger.debug("Initiating key bundle for issuer: %s" % issuer)
+        logger.debug("Initiating key bundle for issuer: %s" % issuer_id)
 
-        if replace or issuer not in self.issuer_keys:
-            self.issuer_keys[issuer] = []
+        _issuer = self.return_issuer(issuer_id)
+        if replace:
+            _issuer.set([])
 
         if jwks_uri:
-            self.add_url(issuer, jwks_uri)
+            _issuer.add_url(jwks_uri)
         elif jwks:
             # jwks should only be considered if no jwks_uri is present
             _keys = jwks['keys']
-            self.issuer_keys[issuer].append(self.keybundle_cls(_keys))
+            _issuer.add_kb(self.keybundle_cls(_keys))
+
+        self[issuer_id] = _issuer
 
-    def find(self, source, issuer):
+    @deprecated_alias(issuer='issuer_id', owner='issuer_id')
+    def find(self, source, issuer_id=None):
         """
         Find a key bundle based on the source of the keys
 
         :param source: A source url
-        :param issuer: The issuer of keys
-        :return: A :py:class:`oidcmsg.key_bundle.KeyBundle` instance or None
-        """
-        try:
-            for kb in self.issuer_keys[issuer]:
-                if kb.source == source:
-                    return kb
-        except KeyError:
-            return None
+        :param issuer_id: The issuer of keys
+        :return: List of :py:class:`oidcmsg.key_bundle.KeyBundle` instances or None
+        """
+        if issuer_id is None:
+            res = {}
+            for _, _issuer in self._issuers.items():
+                kbs = _issuer.find(source)
+                if kbs:
+                    res[_issuer.name] = kbs
+        else:
+            _issuer = self._get_issuer(issuer_id)
+            if _issuer is None:
+                return []
```

Comment:
The old code return None if the issuer was not found, we should probably do the same here

Reply:
Changed this. Are we done now or have I missed something ?

Reply:
I think we're good.